main.cpp

```diff
@@ -58,26 +85,54 @@ struct House
     bool flatRoof = true;
 
     // 1) open and close front door
-    bool operateFrontDoor ();
+    bool operateFrontDoor (bool isFrontDoorOpen);
     // 2) switch on burglar alarm
     bool activateAlarm (int zones);
     // 3) set interior temporature thermostat
     bool setThermostat (float temperatureSetting);
 };
 
-/*
-Thing 2)    Computer
-5 properties:
-    1) Does it have a mouse
-    2) Resolution of display
-    3) How much RAM
-    4) Number of processor cores
-    5) Keyboard or tablet
-3 things it can do:
-    1) Play game
-    2) Switch on and off
-    3) Connect to internet
- */
+bool House::operateFrontDoor(bool isFrontDoorOpen)
+{
+     if(isFrontDoorOpen == true)
+     {
+        std::cout << "Front Door closed\n";
+        return false;
+     }
+    else
+    {
+        std::cout << "Front Door open\n";
+        return true;
+    }  
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise
- [ ] fix the indentation too.